torch/_dynamo/variables/dicts.py

```diff
@@ -457,3 +459,25 @@ def var_getattr(self, tx, name: str) -> "VariableTracker":
 
     def call_hasattr(self, tx, name: str) -> "VariableTracker":
         return variables.ConstantVariable(hasattr(self.obj, name)).add_options(self)
+
+
+def _dictvariable_flatten(d: ConstDictVariable) -> Tuple[List[Any], pytree.Context]:
+    return list(d.items.values()), list(d.items.keys())
+
+
+def _dictvariable_unflatten(
+    values: List[Any], context: pytree.Context
+) -> ConstDictVariable:
+    assert all(isinstance(x, VariableTracker) for x in values)
+
+    return ConstDictVariable(dict(zip(context, values)), user_cls=dict)
```

Comment:
`mutable_local` should be set.

`user_dict` might be wrong since we aren't taking it from the original?

guards?

Reply:
1) Agreed on mutable local
2) Yeah I should enforce user_cls on _dictvariable_flatten
3) I think the ctor created per item guards should suffice for this? `self.guards.update(VariableTracker.propagate(items.values())["guards"])` - which means we don't need to do the `**options` pattern here